code/datums/traits/good.dm

```diff
@@ -96,6 +96,32 @@
 	)
 	H.equip_in_one_of_slots(B, slots , qdel_on_fail = TRUE)
 
+/datum/quirk/multilingual
+	name = "Multilingual"
+	desc = "You spent a portion of your life learning to understand an additional language. You may or may not be able to speak it based on your anatomy."
+	value = 2
+	mob_trait = TRAIT_MULTILINGUAL
+	gain_text = "<span class='notice'>You have learned to understand an additional language.</span>"
+	lose_text = "<span class='danger'>You have forgotten how to understand a language.</span>"
+
+/datum/quirk/multilingual/on_spawn()
+	var/mob/living/carbon/human/H = quirk_holder
+	var/datum/language_holder/LH
+	LH.add_blocked_language(/datum/language/narsie, LANGUAGE_ATOM)
+	LH.add_blocked_language(/datum/language/ratvar, LANGUAGE_ATOM)
+	LH.add_blocked_language(/datum/language/codespeak, LANGUAGE_ATOM)
+	LH.grant_all_languages(TRUE, TRUE, FALSE, LANGUAGE_ATOM)
+	var/datum/language/newLang
+	if(H.job != "Curator")
```

Comment:
You should put all of the above procs under this "if" clause, since there is no point to do any logic if the job is curator.

Reply:
Fair point, I can refactor that quickly enough.

Reply:
Got that rearranged, now procs only run in situations they will be used.